getopt.c

```diff
@@ -0,0 +1,69 @@
+#define _POSIX_C_SOURCE 200809L
+#include <mrsh/getopt.h>
+#include <stdio.h>
+
+char *mrsh_optarg = NULL;
+int mrsh_optind = 1;
+int mrsh_opterr = 1;
+int mrsh_optopt = 0;
+
+int mrsh_getopt(int argc, char *const argv[], const char *optstring) {
+	mrsh_optarg = NULL;
+
+	if (mrsh_optind >= argc) {
+		return -1;
+	}
+
+	if (argv[mrsh_optind] != NULL) {
+		if (argv[mrsh_optind][0] != '-') {
+			return -1;
+		}
+
+		if (argv[mrsh_optind][1] == '\0') {
+			return -1;
+		}
+
+		if (argv[mrsh_optind][1] == '-') {
+			mrsh_optind++;
+			return -1;
+		}
+	}
+
+	const char *c = optstring;
+	if (*c == ':') {
+		c++;
+	}
+
+	mrsh_optopt = 0;
+	int opt = argv[mrsh_optind][1];
+	for (; *c != '\0'; c++) {
+		if (*c != opt) {
+			continue;
+		}
+
+		if (*(c+1) == ':') {
+			if (mrsh_optind + 2 > argc) {
+				mrsh_optopt = opt;
+				if (mrsh_opterr != 0 && optstring[0] != ':') {
+					fprintf(stderr, "%s: Option '%c' requires an argument.\n", argv[0], mrsh_optopt);
```

Comment:
80 cols

Reply:
You could probably make this less nested by breaking out of the loop early if `c[1] != ':'`

Reply:
I generally advocate early return/break, not sure why I've not done it here. I'll see if I can get some time to fix this soon.